File: app/src/main/java/com/mss/imagesearcher/di/FilesModule.java
package com.mss.imagesearcher.di;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.domain.repositories.FilesRepository;
import com.mss.imagesearcher.repositories.files.FilesHelper;
import com.mss.imagesearcher.repositories.files.FilesRepositoryImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class FilesModule {

    @Provides
    @Singleton
    @NonNull
    FilesHelper provideFilesHelper() {
        return new FilesHelper();
    }

    @Provides
    @Singleton
    @NonNull
    FilesRepository provideFilesRepository(FilesHelper filesHelper) {
        return new FilesRepositoryImpl(filesHelper);
    }
}

File: app/src/main/java/com/mss/imagesearcher/di/FilesModule.kt
package com.mss.imagesearcher.di

import com.mss.imagesearcher.model.repositories.FilesRepository
import com.mss.imagesearcher.model.repositories.files.FilesHelper
import com.mss.imagesearcher.model.repositories.files.FilesRepositoryImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class FilesModule {

    @Provides
    @Singleton
    fun provideFilesHelper(): FilesHelper {
        return FilesHelper()
    }

    @Provides
    @Singleton
    fun provideFilesRepository(filesHelper: FilesHelper): FilesRepository {
        return FilesRepositoryImpl(filesHelper)
    }
}
